doc/cookbook/source/examples/regression/least_squares_regression.rst

```diff
@@ -0,0 +1,38 @@
+========================
+Least Squares Regression
+========================
+
+A Linear regression model can be defined as :math:`y_i = \bf{w}.\bf{x_i}` where :math:`y_i` is the predicted value, :math:`\bf{x_i}` is a feature vector and :math:`\bf{w}` is the weight vector. We aim to find the linear function that best explains the data, i.e. minimizes the error or loss function :math:`E(\bf{w})` by finding appropriate :math:`\bf{w}`.
+
+.. math::
+    E({\bf{w}}) = \sum_{i=1}^N(y_i-{\bf w}\cdot {\bf x}_i)^2
+
+One can differentiate :math:`E(\bf{w})` with respect to :math:`\bf{w}` and equate to zero to determine the :math:`\bf{w}` that minimizes :math:`E(\bf{w})`. This leads to solution of the form:
+
+.. math::
+    {\bf w} = \left(\sum_{i=1}^N{\bf x}_i{\bf x}_i^T\right)^{-1}\left(\sum_{i=1}^N y_i{\bf x}_i\right)
+
+-------
+Example
+-------
+
+Imagine we have files with training and test data. We create `CDenseFeatures` (here 64 bit floats aka RealFeatures) and :sgclass:`CRegressionLabels` as
+
+.. sgexample:: least_squares_regression.sg:create_features
+
+We create an instance of the :sgclass:`CLeastSquaresRegression` classifier, passing it training data and labels.
+
+.. sgexample:: least_squares_regression.sg:create_instance
+
+Then we run the `train` function in `LeastSquaresRegression` and apply it to test data.
```

Comment:
never reference function names in examples. Let the code speak for itself. Also function names can change, and we dont want to the rst file to be invariant to such name changes.
On the other hand, the LeastSquaresRegression should be a sgclass reference